### PR TITLE
AE 872: Breakout Agents

### PR DESCRIPTION
**Pull request by rossAE**
Broke agents into separate handlers.

### CODE REVIEW

**Review by arbitrary-sparrow — APPROVED** [review]

File: liquidator/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+The agent-config.json file contains user-configurable agent settings.
+
+- disputePriceErrorPercent: Sets the percent difference between proposed price and actual price that
+  will trigger the AE-UMA-OO-PROPOSEPRICE alert (default value = 0.05, or 5%)
+- cryptoWatchApiKey: API key for Cryptowatch price feeds
+- defipulseApiKey: API key for Defi Pulse price feeds
+- tradermadeApiKey: API key for TraderMade price feeds
+- cmcApiKey: API key for CoinMarketCap price feeds
+
```
Comment (echoAE) [inline liquidator/README.md:30]:
```suggestion
```

File: liquidator/agent-config.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+  "optimisticOracle": {
+    "disputePriceErrorPercent" : 0.05,
+    "cryptowatchApiKey": "",
+    "defipulseApiKey": "",
+    "tradermadeApiKey": "",
+    "cmcApiKey": ""
+  },
```
Comment (echoAE) [inline liquidator/agent-config.json:8]:
```suggestion
```

File: liquidator/package.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "forta-uma-liquidator-bot",
+  "version": "0.0.1",
+  "description": "Collection of scan agent handlers that monitor UMA",
```
Comment (echoAE) [inline liquidator/package.json:4]: Agent that monitors UMA protocol for liquidation opportunities.

File: disputor/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+The agent-config.json file contains user-configurable agent settings.
+
+- disputePriceErrorPercent: Sets the percent difference between proposed price and actual price that
+  will trigger the AE-UMA-OO-PROPOSEPRICE alert (default value = 0.05, or 5%)
+- cryptoWatchApiKey: API key for Cryptowatch price feeds
+- defipulseApiKey: API key for Defi Pulse price feeds
+- tradermadeApiKey: API key for TraderMade price feeds
+- cmcApiKey: API key for CoinMarketCap price feeds
+
```
Comment (echoAE) [inline disputor/README.md:30]: This section of the README.md can be removed for all agents except the optimistic-oracle agent.
```suggestion
```

File: disputor/agent-config.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "optimisticOracle": {
+    "disputePriceErrorPercent" : 0.05,
+    "cryptowatchApiKey": "",
+    "defipulseApiKey": "",
+    "tradermadeApiKey": "",
+    "cmcApiKey": ""
+  },
```
Comment (echoAE) [inline disputor/agent-config.json:8]:
```suggestion
```

File: monitor-mint-calls/README.md [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+The agent-config.json file contains user-configurable agent settings.
+
+- disputePriceErrorPercent: Sets the percent difference between proposed price and actual price that
+  will trigger the AE-UMA-OO-PROPOSEPRICE alert (default value = 0.05, or 5%)
+- cryptoWatchApiKey: API key for Cryptowatch price feeds
+- defipulseApiKey: API key for Defi Pulse price feeds
+- tradermadeApiKey: API key for TraderMade price feeds
+- cmcApiKey: API key for CoinMarketCap price feeds
+
```
Comment (echoAE) [inline monitor-mint-calls/README.md:30]:
```suggestion
```

File: monitor-mint-calls/agent-config.json [hunk 7]
```diff
@@ -0,0 +1,10 @@
+{
+  "optimisticOracle": {
+    "disputePriceErrorPercent" : 0.05,
+    "cryptowatchApiKey": "",
+    "defipulseApiKey": "",
+    "tradermadeApiKey": "",
+    "cmcApiKey": ""
+  },
```
Comment (echoAE) [inline monitor-mint-calls/agent-config.json:8]:
```suggestion
```

File: monitor-mint-calls/package.json [hunk 8]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "forta-uma-mint-call-monitor",
+  "version": "0.0.1",
+  "description": "Collection of scan agent handlers that monitor UMA",
```
Comment (echoAE) [inline monitor-mint-calls/package.json:4]:
```
  "description": "Agent that monitors the UMA protocol for mint() function calls",
```

File: optimistic-oracle/package.json [hunk 9]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "forta-uma-optimistic-oracle",
+  "version": "0.0.1",
+  "description": "Collection of scan agent handlers that monitor UMA",
```
Comment (echoAE) [inline optimistic-oracle/package.json:4]:
```suggestion
  "description": "Agent that monitors the UMA protocol for specific events from the Optimistic Oracle contract",
```

File: disputor/package.json [hunk 10]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "forta-uma-disputor-bot",
+  "version": "0.0.1",
+  "description": "Collection of scan agent handlers that monitor UMA",
```
Comment (echoAE) [inline disputor/package.json:4]:
```suggestion
  "description": "Agent that monitors the UMA protocol for dispute opportunities",
```

File: deployer-watch/package.json [hunk 11]
```diff
@@ -0,0 +1,49 @@
+{
+  "name": "forta-uma-deployer-watch",
+  "version": "0.0.1",
+  "description": "Collection of scan agent handlers that monitor UMA",
```
Comment (echoAE) [inline deployer-watch/package.json:4]:
```suggestion
  "description": "Agent that monitors the UMA protocol for transactions from the deployer address",
```

File: deployer-watch/agent-config.json [hunk 12]
```diff
@@ -0,0 +1,10 @@
+{
+  "optimisticOracle": {
+    "disputePriceErrorPercent" : 0.05,
+    "cryptowatchApiKey": "",
+    "defipulseApiKey": "",
+    "tradermadeApiKey": "",
+    "cmcApiKey": ""
+  },
```
Comment (echoAE) [inline deployer-watch/agent-config.json:8]:
```suggestion
```

File: admin-events/package.json [hunk 13]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "forta-uma-suite",
+  "name": "forta-uma-admin-events",
   "version": "0.0.1",
   "description": "Collection of scan agent handlers that monitor UMA",
```
Comment (echoAE) [inline admin-events/package.json:4]:
```suggestion
  "description": "Agent that monitors the UMA protocol for administrative events on contracts"
```

File: admin-events/README.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+
+The agent-config.json file contains user-configurable agent settings.
+
+- disputePriceErrorPercent: Sets the percent difference between proposed price and actual price that
+  will trigger the AE-UMA-OO-PROPOSEPRICE alert (default value = 0.05, or 5%)
+- cryptoWatchApiKey: API key for Cryptowatch price feeds
+- defipulseApiKey: API key for Defi Pulse price feeds
+- tradermadeApiKey: API key for TraderMade price feeds
+- cmcApiKey: API key for CoinMarketCap price feeds
+
```
Comment (echoAE) [inline admin-events/README.md:30]:
```suggestion
```

**Review by echoAE — APPROVED** [review]
Small changes to streamline some of the `agent-config.json` and `README.md` files.

Small updates to `package.json` files to make description more specific.

The Dockerfile could be simplified for several of the agents, but for the sake of expediency, we will leave them as is for now.

File: liquidator/README.md (hunk 1 above)
Comment (rossAE) [reply liquidator/README.md:30]: done

File: liquidator/agent-config.json (hunk 2 above)
Comment (rossAE) [reply liquidator/agent-config.json:8]: done

File: liquidator/package.json (hunk 3 above)
Comment (rossAE) [reply liquidator/package.json:4]: done

File: disputor/README.md (hunk 4 above)
Comment (rossAE) [reply disputor/README.md:30]: done

File: disputor/agent-config.json (hunk 5 above)
Comment (rossAE) [reply disputor/agent-config.json:8]: done

File: monitor-mint-calls/README.md (hunk 6 above)
Comment (rossAE) [reply monitor-mint-calls/README.md:30]: done

File: monitor-mint-calls/package.json (hunk 8 above)
Comment (rossAE) [reply monitor-mint-calls/package.json:4]: done

File: optimistic-oracle/package.json (hunk 9 above)
Comment (rossAE) [reply optimistic-oracle/package.json:4]: done

File: disputor/package.json (hunk 10 above)
Comment (rossAE) [reply disputor/package.json:4]: done

File: deployer-watch/package.json (hunk 11 above)
Comment (rossAE) [reply deployer-watch/package.json:4]: done

File: deployer-watch/README.md [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+
+The agent-config.json file contains user-configurable agent settings.
+
+- disputePriceErrorPercent: Sets the percent difference between proposed price and actual price that
+  will trigger the AE-UMA-OO-PROPOSEPRICE alert (default value = 0.05, or 5%)
+- cryptoWatchApiKey: API key for Cryptowatch price feeds
+- defipulseApiKey: API key for Defi Pulse price feeds
+- tradermadeApiKey: API key for TraderMade price feeds
+- cmcApiKey: API key for CoinMarketCap price feeds
+
```
[the remark this answers is not in the document]
Comment (rossAE) [reply deployer-watch/README.md:37]: done

File: deployer-watch/agent-config.json (hunk 12 above)
Comment (rossAE) [reply deployer-watch/agent-config.json:8]: done

File: admin-events/package.json (hunk 13 above)
Comment (rossAE) [reply admin-events/package.json:4]: done

File: admin-events/README.md (hunk 14 above)
Comment (rossAE) [reply admin-events/README.md:30]: done

**Pull request merged**